src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -1246,11 +1236,13 @@ def create_python_udf_or_sp(
     statement_params: Optional[Dict[str, str]] = None,
     comment: Optional[str] = None,
     native_app_params: Optional[Dict[str, Any]] = None,
+    runtime_version: Optional[str] = None,
 ) -> None:
-    if session is not None and session._runtime_version_from_requirement:
```

Comment:
this change is to ensure the same value for runtime_version is read from session througout udf/sp registration process.